scripts/pylib/twister/twisterlib.py

```diff
@@ -2045,6 +2051,33 @@ def fill_results_by_status(self):
         for k in self.results:
             self.results[k] = status_to_verdict[self.status]
 
+    def create_build_dir_shortcut(self):
+        """
+        Create build directory with original "long" path. Next take shorter
+        path (based on TestInstance class internal counter) and link them with
+        original path - create shortcut. At the end replace build_dir to
+        shorter path. This action helps to limit path length which can be
+        significant during building by CMake on Windows OS.
+        """
+        if os.name != "nt":  # if OS is not Windows
+            logger.warning("Creating shorter build paths available only on "
+                           "Windows OS")
+            return
+
+        os.makedirs(self.build_dir)
+
+        shortcuts_dir_name = "scs"  # folder for all shortcuts
+        shortcuts_dir_path = os.path.join(self.outdir, shortcuts_dir_name)
+        if not os.path.exists(shortcuts_dir_path):
+            os.makedirs(shortcuts_dir_path)
+
+        shortcut_name = f"sc_{self.shortcut_dir_counter}"
+        shortcut_path = os.path.join(shortcuts_dir_path, shortcut_name)
+        TestInstance.shortcut_dir_counter += 1
+
+        os.system(f"mklink /J {shortcut_path} {self.build_dir}")
```

Comment:
this would be only works in windows.

Reply:
Yes, it will, but I add check if this code is run on Windows at the beginning of this function:
```
def create_build_dir_shortcut(self):
        """
        Create build directory with original "long" path. Next take shorter
        path (based on TestInstance class internal counter) and link them with
        original path - create shortcut. At the end replace build_dir to
        shorter path. This action helps to limit path length which can be
        significant during building by CMake on Windows OS.
        """
        if os.name != "nt":  # if OS is not Windows
            logger.warning("Creating shorter build paths available only on "
                           "Windows OS")
            return
```
`--short-build-path` can be used only on Windows OS (what is described in option's help). Only on Windows OS exists problem with too long directory paths. So, I do not see any sense to extend this option to Linux OS where this problem does not exist.